frontend/spec/support/shared_contexts/checkout_setup.rb

```diff
@@ -25,10 +25,23 @@ def fill_in_address
   def fill_in_credit_card_info(invalid: false)
     fill_in 'name_on_card', with: 'Spree Commerce'
     fill_in 'card_number', with: invalid ? '123' : '4111 1111 1111 1111'
+    check_if_cart_number_valid unless invalid
     fill_in 'card_expiry', with: '12 / 24'
     fill_in 'card_code', with: '123'
   end
 
+  def check_if_cart_number_valid
+    card_number = find_field('card_number').value
+    valid_card_number = '4111 1111 1111 1111'
+
+    return if card_number == valid_card_number
+
+    fill_in "card_number", with: "" if card_number.length > 0
```

Comment:
can we change `"` --> `'` ? let's use `"` only for interpolation

Reply:
My mistake. Will correct it right away!